community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/NodeCorrectlyIndexedCheck.java

```diff
@@ -114,6 +117,24 @@ private void verifyNodeCorrectlyIndexedUniquely(
         }
     }
 
+    private PrimitiveLongIterator filterIfMultipleValuesFound( int propertyKeyId, Object propertyValue,
+            PrimitiveLongIterator indexedNodeIds )
+    {
+        if ( ((PrimitiveLongPeekingIterator) indexedNodeIds).hasMultipleValues() )
```

Comment:
Of course. Thank you, looking at this code in isolation made me blind, I should have looked at where it was used.
It would be an excellent thing to add a comment about in that case.

And since this method is used in only one place, I'd rewrite it to avoid the cast, it's an unnecessary mental stumbling block when reading the code.

(unless of course we find a completely different way to structure this entire change)